ui/analyse/src/study/gamebook/gamebookPlayView.ts

```diff
@@ -92,15 +92,14 @@ function renderFeedback(ctrl: GamebookPlayCtrl, state: State) {
 }
 
 function renderEnd(ctrl: GamebookPlayCtrl) {
-  const study = ctrl.root.study!,
-    nextChapter = study.nextChapter();
+  const study = ctrl.root.study!;
   return h('div.feedback.end', [
-    nextChapter
+    study.nextChapter()
       ? h(
           'a.next.text',
           {
             attrs: dataIcon('G'),
-            hook: bind('click', () => study.setChapter(nextChapter.id)),
+            hook: bind('click', study.goToNextChapter),
```

Comment:
This is the main change. Looks like the hook somehow doesn't get updated correctly? When I do a `console.log` of the chapter here and at the start of `renderEnd`, it correctly gets updated at the start when the chapter changes but when clicking it still reports the old chapter.

Kinda seems like a snapdom bug but since I'm not really familiar with it there might be something else I'm missing or maybe it's intended behavior?

Reply:
Actually, I guess it's simply because changes don't cause a new `onInsert` so it keeps the old event listener.

Reply:
we could force a full vdom re-render by adding a cache-busting class to a top-level element. Studies already do that, by using chapter IDs as classes.

I remember choosing not to use snabbdom's event listeners, but I don't remember why. In any case I don't think it would solve that?

Reply:
I'd assume they would rebind the listener when it changes or something, maybe even just on every redraw. But not sure, maybe it wouldn't fix this, I'm not really familiar with snabbdom.